Sources/PitchSpeller/Data Structures/Pair.swift

```diff
@@ -25,8 +25,39 @@ public struct Pair<A, B, Orderedness> {
 extension Pair: Hashable where A: Hashable, B: Hashable { }
 extension Pair: Equatable where A: Equatable, B: Equatable { }
 
-extension Pair where Orderedness == Ordered {
+extension Pair {
     public var reversed: Pair<B, A, Orderedness> {
         return Pair<B, A, Orderedness>(b, a)
     }
 }
+
+/// map for elements of Pair
+extension Pair {
+    public func map<T, U> (_ aTransform: (A) -> T, _ bTransform: (B) -> U) -> Pair <T, U, Orderedness> {
+        return Pair<T, U, Orderedness> (aTransform(a), bTransform(b))
+    }
+}
+
+/// map for elements of Pair with single transform and elements of same type
+extension Pair where A == B {
+    public func map<T> (_ transform: (A) -> T) -> Pair <T, T, Orderedness> {
+        return Pair<T, T, Orderedness> (transform(a), transform(b))
+    }
+}
+
+public func ==<A, B> (_ lhs: Pair<A, B, Unordered>, _ rhs: Pair<B, A, Unordered>) -> Bool
+    where A: Equatable, B: Equatable {
+    return lhs.a == rhs.b && lhs.b == rhs.a
+}
+
+/// Hashability of Pair where order does not matter
+extension Pair where A: Hashable, A == B, Orderedness == Unordered {
```

Comment:
Very thoughtful!

Reply:
Though, you may still need to declare `Hashable` conformance:

```Swift
extension Pair: Hashable where A: Hashable, A == B, Orderedness == Unordered { ... }
```

Might be smart to add a test like so:

```Swift
func testUnorderedPairHashable {
    let set: Set<Pair<Int,Int,Unordered> = [.init(1,2), .init(2,1)]
    XCTAssertEqual(set.count, 1)
}
```

Reply:
If you don't declare the `Hashable` conformance explicitly, the above _shouldn't_ compile. If `hashValue` is implemented very incorrectly, the test should compile and fail.